.github/workflows/apple-build-checks.yml

```diff
@@ -0,0 +1,102 @@
+name: Apple CI
+
+on:
+  push:
+    branches:
+      - master
+      - next
+    tags:
+      - '**'
+  pull_request: # Run on every pull request, regardless of target branch
```

Comment:
Please avoid automatic triggering of such huge pipelines.
Lets use manual trigger instead:

```
on: workflow_dispatch
```